clients/venice-client/src/main/java/com/linkedin/venice/fastclient/InternalAvroStoreClient.java

```diff
@@ -45,6 +45,50 @@ public final CompletableFuture<Map<K, V>> batchGet(Set<K> keys) throws VeniceCli
     return batchGet(new BatchGetRequestContext<>(keys.size(), false), keys);
   }
 
+  /**
+   * Check for partial failures in the multi-key request context and return the exception if any.
+   * @param multiKeyRequestContext
+   * @return
+   */
+  Throwable checkBatchGetPartialFailure(MultiKeyRequestContext<K, V> multiKeyRequestContext) {
+    Throwable partialResponseException = null;
+    // check for partial failures for multi-key requests
+    if (multiKeyRequestContext.retryContext != null
+        && multiKeyRequestContext.retryContext.retryRequestContext != null) {
+      // retry is triggered
+      if (multiKeyRequestContext.isPartialSuccessAllowed) {
+        if (multiKeyRequestContext.retryContext.retryRequestContext.isCompletedSuccessfullyWithPartialResponse()) {
+          partialResponseException =
+              (Throwable) multiKeyRequestContext.retryContext.retryRequestContext.getPartialResponseException().get();
+        }
```

Comment:
The logic here seems a little confusing.
Is it possible to check whether the streaming response has received all the keys or not as the first step and if not, then try to collect the exception either from the original context or the retry context?
Let us try to converge the logic to always use the partial response API as recommended in the previous comment as the current logic seems to be confusing.

Reply:
I guess `isPartialSuccessAllowed` flag is used as an optimization in case there is no route available for a certain partition, FC can fail fast, but I would argue such type of optimization is not good:
1. The chance of such situation is low.
2. The code becomes more complex and hurts maintainability in some way.

The extra overhead on client side in the exceptional scenario normally is acceptable IMO.

Reply:
removed these newly added logics. 

`RetriableAvroGenericStoreClient->getStreamingCallback->onRecordReceived`,  `onCompletion` and `finalRequestCompletionFuture.whenComplete` does these checks, so just doing a minor change there to account for `retryContext` as well.